internal/cli/resolve.go

```diff
@@ -0,0 +1,254 @@
+// Copyright 2022, 2023 Chainguard, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	apkfs "github.com/chainguard-dev/go-apk/pkg/fs"
+	"github.com/spf13/cobra"
+	"golang.org/x/exp/slices"
+
+	"chainguard.dev/apko/pkg/build"
+	"chainguard.dev/apko/pkg/build/types"
+	"chainguard.dev/apko/pkg/iocomb"
+	"chainguard.dev/apko/pkg/log"
+)
+
+type lock struct {
+	Version  string       `json:"version"`
+	Contents lockContents `json:"contents"`
+}
+
+type lockContents struct {
+	Keyrings     []lockKeyring `json:"keyring"`
+	Repositories []lockRepo    `json:"repositories"`
+	Packages     []lockPkg     `json:"packages"`
+}
+
+type lockPkg struct {
+	Name         string                   `json:"name"`
+	Url          string                   `json:"url"`
+	Version      string                   `json:"version"`
+	Architecture string                   `json:"architecture"`
+	Signature    lockPkg_RangeAndChecksum `json:"signature"`
+	Control      lockPkg_RangeAndChecksum `json:"control"`
+	Data         lockPkg_RangeAndChecksum `json:"data"`
+}
+type lockPkg_RangeAndChecksum struct {
+	Range    string `json:"range"`
+	Checksum string `json:"checksum"`
+}
+
+type lockRepo struct {
+	Name         string `json:"name"`
+	Url          string `json:"url"`
+	Architecture string `json:"architecture"`
+}
+
+type lockKeyring struct {
+	Name string `json:"name"`
+	Url  string `json:"url"`
+}
+
+func resolve() *cobra.Command {
+	var extraKeys []string
+	var extraRepos []string
+	var archstrs []string
+	var output string
+
+	var logPolicy []string
+	var debugEnabled bool
+	var quietEnabled bool
+
+	cmd := &cobra.Command{
+		Use:     "resolve",
+		Short:   "TODO",
+		Long:    `TODO`,
+		Example: `apko resolve <config.yaml>`,
+		Args:    cobra.MinimumNArgs(1),
```

Comment:
We can also mark this as hidden, wdyt?

Reply:
totally. no need to advertise it in the help.